yarn-project/sequencer-client/src/publisher/sequencer-publisher.ts

```diff
@@ -610,48 +612,78 @@ export class SequencerPublisher {
     return true;
   }
 
-  private async getSignalConfig(
-    slotNumber: bigint,
-    signalType: SignalType,
-  ): Promise<{ payload: EthAddress; base: IEmpireBase } | undefined> {
-    if (signalType === SignalType.GOVERNANCE) {
-      return { payload: this.governancePayload, base: this.govProposerContract };
-    } else if (signalType === SignalType.SLASHING) {
-      if (!this.getSlashPayload) {
-        return undefined;
-      }
-      const slashPayload = await this.getSlashPayload(slotNumber);
-      if (!slashPayload) {
-        return undefined;
-      }
-      this.log.info(`Slash payload: ${slashPayload}`);
-      return { payload: slashPayload, base: this.slashingProposerContract };
-    } else {
-      const _: never = signalType;
-      throw new Error('Unreachable: Invalid signal type');
-    }
-  }
-
   /**
-   * Enqueues a castSignal transaction to cast a signal for a given slot number.
+   * Enqueues a governance castSignal transaction to cast a signal for a given slot number.
    * @param slotNumber - The slot number to cast a signal for.
    * @param timestamp - The timestamp of the slot to cast a signal for.
-   * @param signalType - The type of signal to cast.
    * @returns True if the signal was successfully enqueued, false otherwise.
    */
-  public async enqueueCastSignal(
+  public enqueueGovernanceCastSignal(
+    slotNumber: bigint,
+    timestamp: bigint,
+    signerAddress: EthAddress,
+    signer: (msg: TypedDataDefinition) => Promise<`0x${string}`>,
+  ): Promise<boolean> {
+    return this.enqueueCastSignalHelper(
+      slotNumber,
+      timestamp,
+      SignalType.GOVERNANCE,
+      this.governancePayload,
+      this.govProposerContract,
+      signerAddress,
+      signer,
+    );
+  }
+
+  /** Enqueues all slashing actions as returned by the slasher client. */
+  public async enqueueSlashingActions(
+    actions: ProposerSlashAction[],
     slotNumber: bigint,
     timestamp: bigint,
-    signalType: SignalType,
     signerAddress: EthAddress,
     signer: (msg: TypedDataDefinition) => Promise<`0x${string}`>,
   ): Promise<boolean> {
```

Comment:
The `ProposerSlashAction`s are determined by the slasher client, and include voting, creating, and executing slash payloads.